tests/e2e/bin/includes.sh

```diff
@@ -40,97 +40,35 @@ command_exists() {
 	type "$1" > /dev/null 2>&1
 }
 
-check_for_jq() {
-	if $(command_exists "jq"); then
-			return
-	fi
-
-	echo -e $(error_message " jq is not installed. Please install it before moving forward. Instructions: https://stedolan.github.io/jq/download/")
-	exit 1
-}
-
-# check if ngrok is installed
-check_for_ngrok() {
-	if $(command_exists "ngrok"); then
-			NGROK_CMD="ngrok"
-			return
-	fi
-
-	if [ -z "$CI" ]; then
-		echo -e $(error_message "Please install ngrok on your machine. Instructions: https://ngrok.com/download")
-		exit 1
-	fi
-
-	echo -e $(status_message "Installing ngrok in CI...")
-	curl -s https://bin.equinox.io/c/4VmDzA7iaHb/ngrok-stable-linux-amd64.zip > ngrok.zip
-	unzip -o ngrok.zip
-	NGROK_CMD="./ngrok"
-}
-
-# Starts ngrok tunnel. Uses ngrok auth key if available as `$NGROK_KEY`
-start_ngrok() {
-	echo -e $(status_message "Starting ngrok...")
-	# Killing any rogue ngrok instances just in case
-	kill_ngrok
-
-	if [ ! -z "$NGROK_KEY" ]; then
-			$NGROK_CMD authtoken $NGROK_KEY
-	fi
-
-	$NGROK_CMD http -log=stdout 8889 > /dev/null &
-	sleep 3
-	WP_SITE_URL=$(get_ngrok_url)
-
-	if [ -z "$WP_SITE_URL" ]; then
-		echo -e $(error_message "WP_SITE_URL is not set after launching an ngrok")
-		exit 1
-	fi
-}
-
-# Removes all opened tunnels and starts a new one
-restart_ngrok() {
-	echo -e $(status_message "Resetting ngrok...")
-	curl -X "DELETE" localhost:4040/api/tunnels/command_line
-	curl -X "DELETE" "localhost:4040/api/tunnels/command_line%20(http)"
-
-	curl -X POST -H "Content-Type: application/json" -d '{"name":"command_line","addr":"http://localhost:8889","proto":"http"}' localhost:4040/api/tunnels
-
-	sleep 3
-	WP_SITE_URL=$(get_ngrok_url)
-}
+start_env() {
+	yarn wp-env start
 
-get_ngrok_url() {
-	echo $(curl -s localhost:4040/api/tunnels/command_line | jq --raw-output .public_url)
-}
+	# Hacky way to insert PHP code into a wp-config file
+	# It takes care of different tunnel subdomains
+	cat <<EOT > e2e-wp-config-update.sh
+#!/bin/bash
 
-# terminates ngrok process
-kill_ngrok() {
-	ps aux | grep -i ngrok | awk '{print $2}' | xargs kill -9 || true
-}
+touch wp-content/debug.log
+chown www-data:www-data wp-content/debug.log
 
-start_env() {
-	yarn wp-env start
+sed -i "/\/\* That's all, stop editing! Happy publishing. \*\//i\
+define( 'E2E_REQUEST_URL', ( ! empty( \\\$_SERVER['HTTPS'] ) ? 'https://' : 'http://' ) . ( ! empty( \\\$_SERVER['HTTP_HOST'] ) ? \\\$_SERVER['HTTP_HOST'] : 'localhost' ) );\n\
+define( 'WP_SITEURL', E2E_REQUEST_URL );\n\
+define( 'WP_HOME', E2E_REQUEST_URL );\n\
+" wp-config.php
+EOT
 
-	check_for_ngrok
-	check_for_jq
-	start_ngrok
+	yarn wp-env run tests-wordpress sh wp-content/plugins/jetpack-dev/e2e-wp-config-update.sh
```

Comment:
> Is there a reason you don't just put it in tests/e2e/bin

🤦 Looks like I overthought this. There is no reason to have this script in a file :) 

> Also, is there any particular reason not to use `wp config` that I see being used elsewhere?

That might be ideal, but unfortunately, I wasn't able to figure out how to insert arbitrary PHP code via `wp config` command

Reply:
Seems to work with `--raw`, like
```
wp config set E2E_REQUEST_URL "( ! empty( \$_SERVER['HTTPS'] ) ? 'https://' : 'http://' ) . ( ! empty( \$_SERVER['HTTP_HOST'] ) ? \$_SERVER['HTTP_HOST'] : 'localhost' )" --raw
```

Reply:
oh cool! 

It didn't work in this case though. 

since we use docker for both wp install and wp-cli, we pass these wp commands through docker command like so:

`$ yarn wp-env run tests-cli wp config set ....`

I'm not sure why, but I guess that command messes up with shell and `wp-env` escaping and special symbols. I guess there should be a "proper" way to escape it and pass through, but I'm not sure it worth the effort.

Reply:
Looks like wp-env isn't very good about preserving the args when it first concatenates then re-splits the command line. Sigh.

Seems to work if I do it like
```
yarn wp-env run tests-cli "wp config set E2E_REQUEST_URL \"( ! empty( \\\$_SERVER['HTTPS'] ) ? 'https://' : 'http://' ) . ( ! empty( \\\$_SERVER['HTTP_HOST'] ) ? \\\$_SERVER['HTTP_HOST'] : 'localhost' )\" --raw"
```
Passing the whole command as a single argument seems to avoid some extra quote removal somewhere in wp-env. I also got it to work by adding even more quotes (but, oddly, not more backslashes)
```
yarn wp-env run tests-cli wp config set E2E_REQUEST_URL "\"\"( ! empty( \\\$_SERVER['HTTPS'] ) ? 'https://' : 'http://' ) . ( ! empty( \\\$_SERVER['HTTP_HOST'] ) ? \\\$_SERVER['HTTP_HOST'] : 'localhost' )\"\"" --raw
```
Or you could take the easy way and put it in a file in tests/e2e/bin so you can just `yarn wp-env run tests-cli tests/e2e/bin/whatever.sh` and not worry about it.

Reply:
Thanks for digging in! 

I choose the easy way with the `wp-setup.sh` script :)